bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/EGateHandler.java

```diff
@@ -0,0 +1,321 @@
+                        egateSocket.close();
+                        egateSocket = null;
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.toString());
+                    } catch (IOException e1) {
+                        logger.debug("EGate Socket not closed {}", e1.toString());
+                    }
+                    egateSocket = null;
+                }
+                if (egateSocket != null) {
+                    updateStatus(ThingStatus.ONLINE);
+                    startAutomaticRefresh();
+                    logger.debug("EGate Handler started automatic refresh, status: {} ",
+                            getThing().getStatus().toString());
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                    logger.debug("Something went wrong: Host {} Port {}", host, port);
+                }
+            }
+        }
+    }
+
+    private void startAutomaticRefresh() {
+        Runnable runnable = () -> {
+            try {
+                if (reader == null) {
+                    reader = new BufferedReader(new InputStreamReader(egateSocket.getInputStream()));
+                }
+                logger.debug("Socket State: {} to: {}", egateSocket.isConnected(), egateSocket.toString());
+                if (reader.ready()) {
+                    String input = reader.readLine();
+                    logger.debug("Reader got from EGATE: {}", input);
+                    onData(input);
+                }
+            } catch (IOException e) {
+                logger.debug("Error while reading command from Dominoswiss eGate Server {} ", e.toString());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, e.toString());
+            }
+        };
+        refreshJob = scheduler.submit(runnable);
+    }
+
+    /**
+     * Finds and returns a child thing for a given UID of this bridge.
+     *
+     * @param uid uid of the child thing
+     * @return child thing with the given uid or null if thing was not found
+     */
+    public @Nullable Thing getThingByUID(ThingUID uid) {
+        Bridge bridge = getThing();
+
+        List<Thing> things = bridge.getThings();
+
+        for (Thing thing : things) {
+            if (thing.getUID().equals(uid)) {
+                return thing;
+            }
+        }
+
+        return null;
+    }
+
+    @SuppressWarnings("null")
+    protected void onData(String input) {
+        // Instruction=2;ID=19;Command=1;Value=0;Priority=0;
+        Map<String, String> map = new HashMap<String, String>();
+        // split on ;
+        String[] parts = input.split(";");
+        if (parts.length >= 2) {
+            for (int i = 0; i < parts.length; i += 2) {
+                map.put(parts[i], parts[i + 1]);
+            }
+        }
+        // only use FSSReceive Commands
+        /*
+         * Will be used in a Future Release
```

Comment:
Commented out code shouldn't be part of the binding. You can use `git stash` to save the code for further releases.